zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -63,85 +61,52 @@ pub(super) const INDEX_ACCOUNTS_UIVK: &str =
 pub(super) const INDEX_HD_ACCOUNT: &str =
     r#"CREATE UNIQUE INDEX hd_account ON accounts (hd_seed_fingerprint, hd_account_index)"#;
 
-/// Stores diversified Unified Addresses that have been generated from accounts in the
-/// wallet.
+/// Stores addresses that have been generated from accounts in the wallet.
+///
+/// ### Columns
 ///
-/// - The `cached_transparent_receiver_address` column contains the transparent receiver component
-///   of the UA. It is cached directly in the table to make account lookups for transparent outputs
-///   more efficient, enabling joins to [`TABLE_TRANSPARENT_RECEIVED_OUTPUTS`].
+/// - `account_id`: the account whose IVK was used to derive this address.
+/// - `diversifier_index_be`: the diversifier index at which this address was derived.
+/// - `key_scope`: the key scope for which this address was derived.
+/// - `address`: The Unified, Sapling, or transparent address. For Unified and Sapling addresses,
+///   only external-key scoped addresses should be stored in this table; for purely transparent
+///   addresses, this may be an internal-scope (change) address, so that we can provide
+///   compatibility with HD-derived change addresses produced by transparent-only wallets.
+/// - `transparent_child_index`: the diversifier index, if it is in the range of a non-hardened
+///   transparent address index. This is used for gap limit handling and is always populated if the
+///   diversifier index is in that range; since the diversifier index is stored as a byte array we
+///   cannot use SQL integer operations on it and thus need it as an integer as well.
```

Comment:
```suggestion
/// - `transparent_child_index`: the diversifier index in integer form, if it is in the range of a  `u31`
///   (i.e. a non-hardened transparent address index). It is used for gap limit handling, and is set
///   whenever a transparent address at a given index should be scanned at receive time. This
///   includes:
///   - Unified Addresses with transparent receivers (at any valid index).
///   - Unified Addresses without transparent receivers, but within the gap limit of potential
///     sequential transparent addresses.
///   - Transparent change addresses.
///   - ZIP 320 ephemeral addresses.
///   This column exists because the diversifier index is stored as a byte array, meaning that we
///   cannot use SQL integer operations on it for gap limit calculations, and thus need it as an
///   integer as well.
```